lib/seed_data/topics.rb

```diff
@@ -0,0 +1,191 @@
+            name: post.topic.title,
+            selected: unchanged?(post)
+          }
+        end.compact
+      end
+    end
+
+    private
+
+    def topics(include_welcome_topics = true)
+      staff_category = Category.find_by(id: SiteSetting.staff_category_id)
+
+      topics = [
+        # Terms of Service
+        {
+          site_setting_name: 'tos_topic_id',
+          title: I18n.t('tos_topic.title'),
+          raw: I18n.t('tos_topic.body',
+                      company_name: setting_value('company_name'),
+                      base_url: Discourse.base_url,
+                      contact_email: setting_value('contact_email'),
+                      governing_law: setting_value('governing_law'),
+                      city_for_disputes: setting_value('city_for_disputes')
+          ),
+          category: staff_category,
+          static_first_reply: true
+        },
+
+        # FAQ/Guidelines
+        {
+          site_setting_name: 'guidelines_topic_id',
+          title: I18n.t('guidelines_topic.title'),
+          raw: I18n.t('guidelines_topic.body', base_path: Discourse.base_path),
+          category: staff_category,
+          static_first_reply: true
+        },
+
+        # Privacy Policy
+        {
+          site_setting_name: 'privacy_topic_id',
+          title: I18n.t('privacy_topic.title'),
+          raw: I18n.t('privacy_topic.body'),
+          category: staff_category,
+          static_first_reply: true
+        }
+      ]
+
+      if include_welcome_topics
+        # Welcome Topic
+        topics << {
+          site_setting_name: 'welcome_topic_id',
+          title: I18n.t('discourse_welcome_topic.title'),
+          raw: I18n.t('discourse_welcome_topic.body', base_path: Discourse.base_path),
+          after_create: proc do |post|
+            post.topic.update_pinned(true, true)
+          end
+        }
+
+        # Lounge Welcome Topic
+        if lounge_category = Category.find_by(id: SiteSetting.lounge_category_id)
+          topics << {
+            site_setting_name: 'lounge_welcome_topic_id',
+            title: I18n.t('lounge_welcome.title'),
+            raw: I18n.t('lounge_welcome.body', base_path: Discourse.base_path),
+            category: lounge_category,
+            after_create: proc do |post|
+              post.topic.update_pinned(true)
+            end
+          }
+        end
+
+        # Admin Quick Start Guide
+        topics << {
+          site_setting_name: 'admin_quick_start_topic_id',
+          title: DiscoursePluginRegistry.seed_data['admin_quick_start_title'] || I18n.t('admin_quick_start_title'),
+          raw: admin_quick_start_raw,
+          category: staff_category
+        }
+      end
+
+      topics
+    end
+
+    def create_topic(site_setting_name:, title:, raw:, category: nil, static_first_reply: false, after_create: nil)
+      topic_id = SiteSetting.send(site_setting_name)
+      return if topic_id > 0 || Topic.find_by(id: topic_id)
```

Comment:
Our current topic seeding doesn't recreate missing ToS, FAQ and Privacy Policy topics and I kept that behavior. Should we force the creation of those topics if they somehow got deleted and the site settings point to non-existent topics? We had such a case a week ago: https://meta.discourse.org/t/how-to-regenerate-faq-and-tos-pages/110625